cpp/test/util/bitset.cu

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../test_utils.cuh"
+
+#include <raft/random/rng.cuh>
+#include <raft/util/bitset.cuh>
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <numeric>
+
+namespace raft::utils {
+
+struct test_spec {
+  int bitset_len;
+  int mask_len;
+  int query_len;
+};
+
+auto operator<<(std::ostream& os, const test_spec& ss) -> std::ostream&
+{
+  os << "bitset{bitset_len: " << ss.bitset_len << ", mask_len: " << ss.mask_len << "}";
+  return os;
+}
+
+template <typename T>
+void add_cpu_bitset(std::vector<uint32_t>& bitset, const std::vector<T>& mask_idx)
```

Comment:
Would love to see all this host-side logic encapsulated for use elsewhere by making the bitset buildable from host/device memory.